### PR TITLE
support for dhcp ntp server (option 42)

### LINKED ISSUE #2651: Add support for DHCP NTP server
Please add support for ntp server over dhcp.

### PR DESCRIPTION
**Pull request by 532910**
Fixes #2651

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

support for ntp server offered via dhcp (option 42)

### CODE REVIEW

**Comment by c-e-p-x-u-o** [comment]
I'm not sure if ```struct netif * eagle_lwip_getif(uint8 index);``` is correct in sntp.c, but I've not found any other way to define it there.

File: app/modules/sntp.c [hunk 1]
```diff
@@ -805,15 +809,23 @@ static int sntp_sync (lua_State *L)
...
     lua_newtable(L);
-    int i;
-    for (i = 0; i < 4; i++) {
-      lua_pushnumber(L, i + 1);
-      char buf[64];
-      c_sprintf(buf, "%d.nodemcu.pool.ntp.org", i);
-      lua_pushstring(L, buf);
-      lua_settable(L, -3);
+    struct netif *iface = (struct netif *)eagle_lwip_getif(0x00);
+    ip_addr_t ntp_addr = iface->dhcp->offered_ntp_addr;
```
Comment (pjsg) [inline app/modules/sntp.c]: Is this always guaranteed to be set? What happens if the interface was manually configured an didn't use dhcp?

File: app/modules/sntp.c [hunk 2]
```diff
@@ -48,6 +50,8 @@
 #include "rtc/rtctime.h"
 #endif
 
+struct netif * eagle_lwip_getif(uint8 index);
```
Comment (c-e-p-x-u-o) [inline app/modules/sntp.c]: I'm not sure this is correct, but I haven't found another way to define it here.

File: app/modules/sntp.c (hunk 1 above)
Comment (c-e-p-x-u-o) [reply app/modules/sntp.c]: Is this a trick question?
The real answer is "I don't know and havn't tested this case". looks like it should be: 
```
ip_addr_t ntp_addr = iface->dhcp ? iface->dhcp->offered_ntp_addr : NULL;
```

**Comment by marcelstoer** [comment]
Superseded by #2709

**Pull request closed without merging**